UnityProject/Assets/Scripts/Fire/FireSource.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Defines fire sources like candles, lighters or activated welder
+/// Fire source can lead to plasma fire or used to ignite flamable objects
+/// </summary>
+public class FireSource : MonoBehaviour, IServerSpawn
+{
+	[SerializeField]
+	[Tooltip("Does this fire emit flame from start?")]
+	private bool isBurningOnSpawn;
+
+	[Temperature]
+	[SerializeField]
+	[Tooltip("Flame temperature in Kelvins")]
+	private float flameTemperature = 700f;
+
+	[SerializeField]
+	[Tooltip("Volume of flamed gas in m3")]
+	public float flameVolume = 0.005f;
+
+	private PushPull pushPull;
+
+	/// <summary>
+	/// Does this object emit flame?
+	/// Flame may lead to ignite flamable objects or plasma fire
+	/// </summary>
+	public bool IsBurning { get; set; }
+
+	private void Awake()
+	{
+		pushPull = GetComponent<PushPull>();
+	}
+
+	private void Update()
```

Comment:
I would suggest using `UpdateManager` to handle updates. That way you can subscribe for updates only when needed - for example when it is the server and `pushPull` is not null, and `IsBurning`.

Reply:
Yep, implemented that. Didn't touch Cigarette and Lighters update loops, cause they will refactored soon for new sprite system.